### ISSUE TITLE
Validation Requires recipient-country On Activity When Countries Present On Transactions

### ISSUE THREAD

**Issue by iatisupport** [younginnovations/aidstream-v201-migration#7]
I created an activity that had transactions which each contained a recipient-country field. however, when I came to mark the activity as 'Completed' the validation still required me to also add a recipient-country or recipient-region at activity level. However the rule at v2.01 is that if countries or regions are present on transactions they must not be present at the activity level as well - http://iatistandard.org/201/upgrades/integer-upgrade-to-2-01/migrating/#additional-elements-in-transactions-sector-recipient-country-recipient-region

My test activity is 'ZZ test Activity' in demo account

**Comment by anjesh** [comment]
In this case, then we need to make sure that the countries or regions are present in the transactions only and not in the activity-level country, and shouldn't allow users to enter activity-level region/country. This requires complex validation in each activity. 

What if one activity transaction has country and other transaction of the same activty doesn't have. Should we stop users from "complete activity", until all the transactions have country/region.

**Comment by iatisupport** [comment]
Yes if transactions (or regions) are specified at transaction level then it must be present on all transactions - http://iatistandard.org/201/activity-standard/overview/transactions/

**Comment by anjesh** [comment]
Here's the scenario. 

Scenario 1:
Activity has 2 transactions, and one transaction has transaction-recipient-country. Activity also has activity-recipient-country

When the user tries to "complete", the following error should be displayed:
`You may either have activity-recipient-country or choose to have recipient-country for all your transactions. Please remove transaction-recipient-country to complete it. Or delete activity-recipient-country and add transaction-recipient-country to all your transactions. `

Scenario 2:
Activity has 2 transactions, and one transaction has transaction-recipient-country. Activity doesn't have  activity-recipient-country.

When the user tries to "complete", the following error should be displayed:
`You need to add transaction-recipient-country to all your transactions.Or you may choose to add activity-recipient-country and delete all of your transaction-recipient-country to complete your activity.`

The validation is little tricky here and confusing for the users. Thoughts? @iatisupport

**Comment by iatisupport** [comment]
I agree that this is complicated validation but I would suggest that we go with the validation as listed above in the 2.01 version of Aidstream and perhaps look at how this can be better handled via the interface of the 'new' Aidstream also currently in development. Perhaps in the new Aidstream if the user specifies a country at the activity level they could be prompted to acknowledge that they will therefore not be able to add recipient-country at the transaction level?

**Comment by anjesh** [comment]
Now extending this validation to address #15 

* User adds activity-region and activity-country and doesn't write % or % doesn't sum to 100. User doesn't add transaction-country/region.

`You have to mention % when adding both activity-region and activity-country and % should sum up to 100%.`

* User adds activity-region and activity-country and doesn't write % or % doesn't sum to 100. User also adds transaction-country/region in one of the transactions. 

`You have to mention % when adding both activity-region and activity-country and % should sum up to 100%. Also you can't have transaction-country/region when you have activity-region and/or activity-country. You may choose to remove activity-region and activity-country and only add transaction-country/region`

* User doesn't add activity-region and activity-country. User has transaction-country/region in one of the transactions. 

`You need to add transaction-recipient-country/region to all your transactions.Or you may choose to add activity-recipient-country or recipient-region and delete all of your transaction-recipient-country to complete your activity.`
